### DISCUSSION TITLE
Tabindex requires two tabs to focus each component in reactive form.

### DISCUSSION THREAD

**Discussion by mdrizwanahmad1** [primefaces/community#1734] [PrimeNG]
I am using primeNG 12.0 and if i define the tabindex for component like inputSwitcher, radiobutton, calendar, when i hit two time then focus is going to selection mode, 

First focus goes to outer layer of component and then again hit tab it goes to selection places.

Expected. Once hit tab it should focus to select a component.
<div [formGroup]="settingForm" >
   <div>
	<p-inputSwitch tabindex="2" onChange="onChangeEvent($event)"formControlName="inputSwitcherName">
	<p-inputSwitch>

   </div>
</div>

Note: I have one component outside of form which i have defined tabindex="1", that works fine

Please reply if anyone faces this kind of the issue, or do we have any suitable solutions for this

**Comment by cetincakiroglu** [comment]
Hi,

PrimeNG 12 doesn't have accessibility support. Could you please use the v17? Also, could you please share a stackblitz example so we can better understand the issue?

**Reply by mdrizwanahmad1** [reply]
Below are the example, first div has one input component with tabindex=1 this works proper second has div with form which has inputswich component when i press tab then it goes to p-inputswich component outer layer then again need to press to select inputswich, i have also attached screenshots for inputSwitch two tab focus state


div
 input ([ngModel])=="title" tabindex="1"
 /input
/div
div [formgroup]= "reactivePocFormGroup"
div
           p-inputSwich tabindex="2" formControlName="inputSwitchControl"
           /p-inputSwich 
   
/div 
/div 

<img width="161" alt="focus_issue" src="https://github.com/primefaces/community/assets/167769809/af265988-0ce3-4203-8690-0a22bdfb2666">

**Comment by mdrizwanahmad1** [comment]
Another example is below where tabindex is not working based on tabindex,
We have reactive form in pocreactiveform.html which we are using in app.component.html, 

div
    input [(ngModel)]="title" autofocus tabindex="1"/
/div
 app-pocreactiveform></app-pocreactiveform


Pocreactiveform we have below code 

div [formGroup]="settingForm"
  input formControlName="title1" tabindex="3"/
  p-button  icon="pi pi-external-link" label="test" tabindex="2"
/p-button
/div
p-button  icon="pi pi-external-link" label="submit" tabindex="10"
/p-button

when i press tab it goes via tab 1 then next focus goes to tabindex 3 then tabindex2 so on
expected is it should o via tabindex1 then 2 then 3
